### ISSUE TITLE
PS3 support

### ISSUE THREAD

**Issue by r3sus** [garyttierney/ghidra-batteries-included#5]
Hello and thanks a lot for your work!
That'd be very useful to add PS3 .ELF support in Ghidra.
I tried various scripts and add-ons to Ghidra but got only issues:
There are my findings:
1.  [clienthax/Ps3GhidraScripts](https://github.com/clienthax/Ps3GhidraScripts)
[issue](https://github.com/clienthax/Ps3GhidraScripts/issues/1)

2.  [zecoxao/ps3_ghidra](https://github.com/zecoxao/ps3_ghidra)
a. CELL PPU Processor Module
b. find_stdu.java
[issue](https://github.com/zecoxao/ps3_ghidra/issues/1)

I didn't tried:
https://github.com/aerosoul94/GhidraSPU
:> https://github.com/bevanweiss/GhidraSPU/tree/TrialChanges

**Comment by garyttierney** [comment]
Next build will include the PPU and SPU sleigh processor modules.

**Comment by r3sus** [comment]
Awesome, thank you!
Xbox360 support will be useful too, but doesn't supports 9.2 now
https://github.com/zeroKilo/XEXLoaderWV/issues/8

**Comment by clienthax** [comment]
@garyttierney To avoid issues with decompliation on ps3 the following change is needed in Processors\PowerPC\data\languages\ppc_64_32.cspec
Add <register name="r2"/> to the <unaffected> list.

Apart from that my scripts just need the indexes bumping due to a header change in ghidra.. (Why the heck don't they allow me to lookup things by name)

**Comment by garyttierney** [comment]
Hi @clienthax,

>  the following change is needed in Processors\PowerPC\data\languages\ppc_64_32.cspec
Add to the list.

Is this a commit you're referring to? I think I have a copy of your PPC64 SLEIGH changes from maybe a year or so ago, but it seems to be missing from your repo now.

**Comment by clienthax** [comment]
> Hi @clienthax,
> 
> > the following change is needed in Processors\PowerPC\data\languages\ppc_64_32.cspec
> > Add to the list.
> 
> Is this a commit you're referring to? I think I have a copy of your PPC64 SLEIGH changes from maybe a year or so ago, but it seems to be missing from your repo now.

Gah.. copypasta fail!

add r2 to the <unaffected> list

**Comment by garyttierney** [comment]
Ah, got it. I'll include a patch for that in my next PR. Is there any reason not to send that upstream to Ghidra?

**Comment by clienthax** [comment]
> Ah, got it. I'll include a patch for that in my next PR. Is there any reason not to send that upstream to Ghidra?

I think it's been submitted.. somewhere, likely by aero

**Comment by garyttierney** [comment]
Yep, found it. For posterity: https://github.com/NationalSecurityAgency/ghidra/issues/570#issuecomment-701728185.

I'll open a PR for this.

**Comment by garyttierney** [comment]
Patch included here: https://github.com/garyttierney/ghidra-batteries-included/pull/28. I'll keep this open 'til I bundle the SELF analyzer script and the sleigh module for the SPU.

**Comment by PikminGuts92** [comment]
Any update on this?

**Comment by clienthax** [comment]
@garyttierney my scripts are now an extension https://github.com/clienthax/Ps3GhidraScripts/releases
